### PR TITLE
[#ICC-115 #ICC-104] Add EVH Env variables to Messages CQRS

### PR DESCRIPTION
**Pull request by AleDore**
### List of changes
- Add Payment updater EVH ref on CQRS messages
- Add env variables

### Motivation and context
This PR is needed by https://github.com/pagopa/io-functions-messages-cqrs/pull/12/files

### Type of changes

- [ ] Add new resources
- [x] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [x] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [ ] No

### Other information

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-infra.deploy](https://dev.azure.com/pagopaspa/io-iac-projects/_build?definitionId=218)
2. select PR branch
3. wait for approval

### CODE REVIEW

**Comment by gunzip** [comment]
[Evh](https://it.wikipedia.org/wiki/Eddie_van_Halen) :metal:

File: src/core/function_messages_cqrs.tf [hunk 1]
```diff
@@ -85,6 +108,10 @@ module "function_messages_cqrs" {
       "AzureWebJobs.CosmosApiMessageStatusChangeFeedForView.Disabled" = "0"
       "AzureWebJobs.HandleMessageViewUpdateFailures.Disabled"         = "0"
       "AzureWebJobs.UpdateCosmosMessageView.Disabled"                 = "0"
+      "AzureWebJobs.UpdatePaymentOnMessageView.Disabled"              = "0"
+      "AzureWebJobs.HandlePaymentUpdateFailures.Disabled"             = "0"
+      "AzureWebJobs.CosmosApiMessagesChangeFeed.Disabled"             = "0"
+      "AzureWebJobs.HandleMessageChangeFeedPublishFailures.Disabled"  = "0"
```
Comment (fabriziopapi) [inline src/core/function_messages_cqrs.tf:114]: We must remember to manually mark the environament viariable as slot-dependent on the azure developer portal

File: src/core/data.tf [hunk 2]
```diff
@@ -184,3 +184,18 @@ data "azurerm_storage_account" "cdnassets" {
   name                = "iopstcdnassets"
   resource_group_name = var.common_rg
 }
+
+data "azurerm_eventhub" "payment_updater_evh" {
```
Comment (fabriziopapi) [inline src/core/data.tf:188]: @pasqualedevita we referenced an event hub defined somewhere else in this way. Is it the right way to do it?

Comment (pasqualedevita) [reply src/core/data.tf:188]: yes it's ok :)

File: src/core/function_messages_cqrs.tf (hunk 1 above)
Comment (pasqualedevita) [reply src/core/function_messages_cqrs.tf:114]: since this function has only async operations probably we can dismantle the staging slot

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
errors on data

File: src/core/function_messages_cqrs.tf [hunk 3]
```diff
@@ -14,16 +14,39 @@ locals {
...
+      TARGETKAFKA_transactionalId     = "IO_MESSAGES_CQRS"
+      TARGETKAFKA_topic               = "messages-payments"
+
+      PAYMENT_FOR_VIEW_TOPIC_NAME                       = "payment-updates"
+      PAYMENT_FOR_VIEW_TOPIC_CONSUMER_GROUP             = "$Default"
+      PAYMENT_FOR_VIEW_TOPIC_CONSUMER_CONNECTION_STRING = data.azurerm_eventhub.payment_updater_evh.keys["payment-updates.io-fn-messages-cqrs"].primary_connection_string
+
+      APIM_BASE_URL = "https://api-internal.io.italia.it"
+      APIM_SUBSCRIPTION_KEY = data.azurerm_key_vault_secret.apim_services_subscription_key.value
+
```
Comment (gquadrati) [inline src/core/function_messages_cqrs.tf:49]: It seems `PN_SERVICE_ID` is missing here

File: src/core/function_messages_cqrs.tf [hunk 4]
```diff
@@ -14,16 +14,39 @@ locals {
...
       COSMOSDB_CONNECTION_STRING = format("AccountEndpoint=%s;AccountKey=%s;", data.azurerm_cosmosdb_account.cosmos_api.endpoint, data.azurerm_cosmosdb_account.cosmos_api.primary_master_key)
 
-      MESSAGE_VIEW_UPDATE_FAILURE_QUEUE_NAME = "message-view-update-failures"
-      MESSAGE_CONTAINER_NAME                 = "message-content"
-      MESSAGE_CONTENT_STORAGE_CONNECTION     = data.azurerm_storage_account.api.primary_connection_string
-      QueueStorageConnection                 = data.azurerm_storage_account.api.primary_connection_string
+      MESSAGE_VIEW_UPDATE_FAILURE_QUEUE_NAME          = "message-view-update-failures"
+      MESSAGE_VIEW_PAYMENT_UPDATE_FAILURE_QUEUE_NAME  = "message-view-paymentupdate-failures"
+      MESSAGE_PAYMENT_UPDATER_FAILURE_QUEUE_NAME      = "message-paymentupdater-failures"
+      MESSAGE_CONTAINER_NAME                          = "message-content"
```
Comment (gquadrati) [inline src/core/function_messages_cqrs.tf:20]: We should create these queues, and also the related poison queues

**Review by gquadrati — CHANGES_REQUESTED** [review]

**Review by pasqualedevita — COMMENTED** [review]
@AleDore I can't see the poison queues....

**Review by pasqualedevita — DISMISSED** [review]

**Review by pasqualedevita — APPROVED** [review]

**Pull request merged**